### DISCUSSION TITLE
How to obtain the daily kw energy consumption

### DISCUSSION THREAD

**Discussion by daveh24706** [raomin/ESPAltherma#393] [Q&A]
Apologies for asking a stupid question, ive everything running well, receiving data from my altherma, just wondering from what attributes can I calculate the hourly, daily kw energy consumption. It's the only bit of data im missing on my dashboard. my altherma already displays the hourly energy consumption on the display, just wondering how I can get this on home assistant. do I need to do a calculation ie voltage current etc?

Ideally what id love to display on home assistant is the kw consumption x my electricity cost kw per hour rate.

**Comment by redtree99** [comment]
I've just looked at this and think I've got it.

First create a sensor that tracks kW.  This is my code:
      - name: "INV secondary power (kW)"
        unique_id:  "INV secondary power (kW)"
        unit_of_measurement: 'kW'
        device_class: current
        state_class: measurement
        state: >
            {{ state_attr('sensor.none_althermasensors', 'INV secondary current (A)')*242/1000 }}

The 242 is the local voltage, you may be able to get that from the Daikin unit but it isn't available on mine.  Dividing by 1000 converts it to kW.  It is just the simple equation Amps x Volts = Watts 

https://www.webstaurantstore.com/guide/600/how-to-calculate-amps-volts-and-watts.html

Then use Utility meter to track this new sensor: https://www.home-assistant.io/integrations/utility_meter/

Only just done this but so far I think it works

**Reply by ocsele** [reply]
I was trying to do the same, but couldn't decide which pump's usage should I use, so I ended up calculating both, and adding them up, but it seems to be higher, then what I draw from the power line (I'm reading my smart meter). Should I only use secondary pump's current? If yes, care to explain to me, why that one, and what's the difference between primary and secondary?

**Reply by redtree99** [reply]
Definitely use the primary one. I’ve installed a Shelly power meter and the readings are virtually the same as the primary one. Not sure what the secondary one is but it doesn’t correlate with the actual energy usage. When I have time I’m going to compare the Shelly vs the Altherma Primary in detail but I think the main difference will be changes in voltage that affect power consumption. That  shows up on the Shelly but in my Altherma code I’ve fixed the voltage at 242v as I can’t get voltage from the Altherma.

**Reply by redtree99** [reply]
I think the secondary one is the current of a heat pump component (pump?) that is running at a much different voltage so using 242v gives an inaccurate kW figure

**Reply by ocsele** [reply]
Thx, a lot Redtree!

Started to track both pump's calculated consumption, secondary (light green) seems way higher than primary (dark green). As you can see, the grey part of the bars are the "untracked" or "unmeasured" oart of my total consumption (that is not assigned to individual devices in my household), and the untracked is still a bigh chunk of it (I don't really have any other big consumer in my house apart from the heatpump), so I think I have to consider both pump's consumption to get the total heat pump consumption, and it still might under calculate it (I'm using a Reimann sum integral sensor in home assistant with trapezoidal integration method, I'm not sure if this is the absolutely correct way to calculate it..

![kép](https://github.com/user-attachments/assets/0dded98a-e0b4-49ad-a998-9ca9c51b706f)

**Reply by redtree99** [reply]
Definitely remove the secondary one and ignore it.  I’ve tracked the heat pump energy usage with a Shelly EM and it is virtually the same as the Primary one.  If you still have a difference then it might be how you’ve calculated the kWh using the Riemann sum. I started having a play with this but ran out of time so just got the Shelley EM installed instead. Have you checked your calculated figures with the figures directly from the Altherma?  They are within 0.5kWh of the actuals I get from the Shelly EM.  You’ll need to add the heating and hot water ones I’ve circled in red.

![IMG_1228](https://github.com/user-attachments/assets/5212c749-2b36-48bd-8c7f-a6d9aa2e64a5)

**Reply by redtree99** [reply]
Just realised you may not have this info.  You can get it through this: https://github.com/jwillemsen/daikin_onecta

This pulls the info from the Daikin cloud rather than locally

**Reply by ocsele** [reply]
Hmm...
- I don't have any cloud connection, not even sure if it is even an option for me on an altherma3? That's the main reason I'm using espaltherma, to get ANY kind of dataand obviously, and is it seems I do not receive any energy related information over modbus, only current draw
- I'll check if I have energy usage/produce values on the device itself, can you help me guess what should be a rough ratio of consumed and produced energy?
- I was thinking of going the shelly route since I'm already using shelly devices, but only for lights and roller shutters, I was a little afraid of implementing it to such an essential device as a heat pump. Should I not be worried at all? Adding a shelly to the altherma is an easy and safe DIY? Where exactly did you install yours in the line? Perhaps could you share a couple of photos of the install to help me picture it?

many thx in advance!!!!!

**Reply by redtree99** [reply]
I can’t see a reason why you wouldn’t be able to have cloud access if you haven’t already. Have a look here: https://www.daikin.co.uk/en_gb/product-group/control-systems/daikin-online-controller.html


I’ll get some pics of the Shelly but there is nothing to be worried about. It is just a clamp that fits around the power cable to the heat pump so it doesn’t affect the heat pump. I had an electrician fit it but if you have a power socket near to where the supply to the heat pump you could use that.

**Reply by redtree99** [reply]
You should also be able to see energy use on the local MMI. https://community.openenergymonitor.org/uploads/short-url/tfYQT6HIIA9GPAjBYugVYBQvkUh.pdf

**Reply by ocsele** [reply]
I'm sorry, I must have gotten my altherma type wrong, it's an older model with monochrome display, and no LAN interface, so it is definitely lacking cloud functionality.. before tapping into it with espaltherma to read the data I inquired for a LAN module, but it was absolutely not worth the price tag... :(

**Comment by check0r0** [comment]
I really would like to have this function. 
I already tried to get it with the help of chatGPT but it's not very accurate.
